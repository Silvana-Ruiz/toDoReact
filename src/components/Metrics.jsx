import React from 'react'
import useToDoContext from '../hooks/useToDoContext'

const Metrics = () => {
const { metrics } = useToDoContext();
const { averageTimeTotal, averageTimeLowTasks, averageTimeMediumTasks, averageTimeHighTasks } = metrics;
  return (
    <div className='grid grid-cols-2 my-9'>
        <div className='justify-self-center'>
            <p>Average time to finish tasks:</p>
            <p>{averageTimeTotal}</p>
        </div>
        <div className='justify-self-center'>
            <p>Average time to finish tasks by priority:</p>
            <p>Low: {averageTimeLowTasks}</p>
            <p>Medium: {averageTimeMediumTasks}</p>
            <p>High: {averageTimeHighTasks}</p>
        </div>
    </div>
  )
}

export default Metrics