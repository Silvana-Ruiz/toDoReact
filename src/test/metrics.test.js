import React from 'react';
import { render } from '@testing-library/react';
import Metrics from '../components/Metrics';

jest.mock('../hooks/useToDoContext', () => () => ({
  metrics: {
    averageTimeTotal: '0:00',
    averageTimeLowTasks: '0:00',
    averageTimeMediumTasks: '0:00',
    averageTimeHighTasks: '0:00',
  },
}));

describe('Metrics Component', () => {
  it('displays average to do time metrics correctly', () => {
    const { getByText } = render(<Metrics />);
    
    expect(getByText('Average time to finish tasks')).toBeInTheDocument();
    expect(getByText('Average time to finish tasks by priority')).toBeInTheDocument();
  });

  it('matches snapshot', () => {
    const { container } = render(<Metrics />);
    expect(container).toMatchSnapshot();
  });
});
