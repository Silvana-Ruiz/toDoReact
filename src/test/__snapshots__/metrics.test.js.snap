// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Metrics Component matches snapshot 1`] = `
<div>
  <div
    class="grid grid-cols-2 my-10 border bg-white rounded-2xl shadow-inner px-12 py-10 "
  >
    <div
      class="justify-self-center"
    >
      <p
        class="mb-2"
      >
        Average time to finish tasks
      </p>
      <p
        class="text-xl font-bold italic text-neutral-600 text-center"
      >
        0:00
      </p>
    </div>
    <div
      class="justify-self-center"
    >
      <p
        class="text-center mb-2"
      >
        Average time to finish tasks by priority
      </p>
      <div
        class="flex flex-row gap-4 text-center justify-center"
      >
        <div>
          <p
            class="text-xl font-bold italic text-neutral-600"
          >
            0:00
          </p>
          <p
            class="font-medium"
          >
            Low
          </p>
        </div>
        <div>
          <p
            class="text-xl font-bold italic text-neutral-600"
          >
            0:00
          </p>
          <p
            class="font-medium"
          >
            Medium
          </p>
        </div>
        <div>
          <p
            class="text-xl font-bold italic text-neutral-600"
          >
            0:00
          </p>
          <p
            class="font-medium"
          >
            High
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
